engine/src/main/java/org/hibernate/search/query/engine/spi/EntityInfo.java

```diff
@@ -23,9 +22,11 @@
 
 	Object[] getProjection();
 
-	List<Integer> getIndexesOfThis();
+	Iterable<Integer> getIndexesOfThis();
```

Comment:
Also, exposing any form of List wouldn't allow us to have second thoughs on what kind of collection we'd need. In this specific case, I wanted to replace the List with a Set.